fish-rust/src/wutil/format/format.rs

```diff
@@ -514,3 +514,19 @@ impl Printf for &WString {
         self.as_utfstr().format(spec)
     }
 }
+
+fn add_padding(mut s: WString, spec: &ConversionSpecifier) -> Result<WString> {
+    let width: usize = match spec.width {
+        NumericParam::Literal(w) => w,
+        _ => {
+            return Err(PrintfError::Unknown); // should not happen at this point!!
+        }
+    }
+    .try_into()
+    .unwrap_or_default();
+    while s.len() < width {
+        s.insert(0, ' ');
```

Comment:
should use `str::repeat` to avoid quadratic runtime.

Reply:
Done